src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Form/Entity/OidcEntityType.php

```diff
@@ -0,0 +1,458 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Form\Entity;
+
+use Surfnet\ServiceProviderDashboard\Application\Command\Entity\SaveOidcEntityCommand;
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Entity;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\Extension\Core\Type\TextareaType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class OidcEntityType extends AbstractType
+{
+
+    const GRANT_TYPE_AUTHORIZATION_CODE = 'authorization';
+    const GRANT_TYPE_IMPLICIT = 'implicit';
+
+    private static $responseTypes = [
+        Entity::GRANT_TYPE_AUTHORIZATION_CODE_CODE => self::GRANT_TYPE_AUTHORIZATION_CODE,
+        Entity::GRANT_TYPE_IMPLICIT_ID_TOKEN_TOKEN => self::GRANT_TYPE_IMPLICIT,
+        Entity::GRANT_TYPE_IMPLICIT_ID_TOKEN => self::GRANT_TYPE_IMPLICIT,
+    ];
+
+    /**
+     * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+     * @SuppressWarnings(PHPMD.UnusedLocalVariable) - for the nameIdFormat choice_attr callback parameters
+     *
+     * @param FormBuilderInterface $builder
+     * @param array $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            // The first button in a form defines the default behaviour when
+            // submitting the form by pressing ENTER. We add a 'default
+            // action' button on top of the form so the controller action
+            // handling the form submission can choose what action to
+            // perform. This is to prevent the import action when submitting
+            // the form (the import button is now the second button on the
+            // form).
+            ->add('default', SubmitType::class, ['attr' => ['style' => 'display: none']])
+            ->add(
+                $builder->create(
+                    'general',
+                    FormType::class,
+                    [
+                        'inherit_data' => true,
+                        'label' => 'General information',
+                    ]
+                )
+                    ->add(
+                        'ticketNumber',
```

Comment:
This might be a good moment to remove the ticketNumber from the entity forms, @teunfransen is this OK with you? Jira tickets are not saved on the entity forms in production, are they?

Reply:
@MKodde They are not. So, good idea.